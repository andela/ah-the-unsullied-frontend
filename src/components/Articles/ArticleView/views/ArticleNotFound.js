import React from ''; 

const notFound = () => {
   return(
       <div>
       <h1>404 PAGE NOT FOUND</h1>
       </div>
   )
}

export default notFound;