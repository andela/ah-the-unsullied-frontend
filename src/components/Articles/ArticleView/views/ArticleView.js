import { Row, Col } from 'react-materialize';
import { NavLink } from 'react-router-dom';
import dateformat from 'dateformat';
import renderHTML from 'react-render-html';
import React from 'react';
import '../../../../assets/css/articleDetail.scss';

const articleDetail = props => {
  const article = props.article.article.article;
  const author = article.author;
  const followStatus = author.following;
  const tags = article.tag_list;
  const date = article.created_at;
  const formatDate = dateformat(date, 'dddd, mmmm dS, yyyy');

  const getTags = () =>
    tags.map(item => {
      return <button className="tags-btn">{item}</button>;
    });

  const getImage = () => {
    if (!author.image) {
      const new_image = require('../../../../assets/images/profile.png');
      return new_image;
    } else {
      return author.image;
    }
  };

  const followBtn = () => {
    if (followStatus === null) {
      return (
        <div>
          <button className="follow">Follow</button>
        </div>
      );
    }
  };

  const ownerUpdate = () => {
    const auth = props.auth;
    if (auth.isAuthenticated) {
      const article = props.article.article.article;
      const author = article.author;
      const user = auth.user.username;

      if (user === author.username) {
        return (
          <div className="update-link">
            <NavLink to="/edit-article">
              <i className="material-icons">edit</i>
            </NavLink>
          </div>
        );
      }
    }
  };

  return (
    <div className="container">
      <div>
        <h3>{article.title}</h3>
        <h5 className="description">{article.description}</h5>
        <Row>
          <Col s={1}>
            <img
              src={getImage()}
              className="responsive-img circle escalate"
              height="50px"
              width="50px"
              alt="Avatar"
            />
          </Col>

          <Col s={1}>
            <a href="/">{author.username}</a>
            {followBtn()}
          </Col>
          <Col s={2}>
            {' '}
            <b>{article.read_time}</b>
          </Col>
          <Col>
            <b className="description">{formatDate}</b>
          </Col>
          { ownerUpdate() }
        </Row>
        <div className="bd-image">
          <p>{renderHTML(article.body)}</p>
        </div>
        {getTags()}

        <hr />
        <div />
      </div>
    </div>
  );
};

export default articleDetail;
