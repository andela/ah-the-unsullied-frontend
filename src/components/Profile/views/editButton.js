import React from 'react';

const viewEditProfile = () => {};

export default viewEditProfile;
