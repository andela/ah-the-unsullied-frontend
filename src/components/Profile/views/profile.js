import React from 'react';
import { Card, CardTitle, Row, Col } from 'react-materialize';

const profile = props => {
  const { profile } = props;
  return (
    <div className="row">
      <div className="col s12">
        <div className="profile">
          <div className="card profile">
            <div className="card-content">
              <div className="row">
                <div className="col s2">
                  <div />
                  <div className="profile_image">
                    <img
                      src={profile.image}
                      className="responsive-img"
                      height="200px"
                      width="200px"
                    />
                  </div>
                </div>
                <div className="bio col s8">
                  <div className="username">{profile.username}</div>
                  <div className="bio col s4">{profile.bio}</div>
                </div>
                <div className="bio col s2">
                  <button className="btn waves-effect edit">
                    Edit profile
                    <i className="material-icons right">edit</i>
                  </button>
                </div>
              </div>
            </div>
            <div className="card-action">
              <a href="#">articles</a>
              <a href="#">highlights</a>
              <a href="#">Followers</a>
              <a href="#">Following</a>
              <a href="#">Ratings</a>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};
export default profile;
