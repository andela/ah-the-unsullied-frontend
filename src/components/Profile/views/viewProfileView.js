import React from 'react';
import { Row, Col } from 'react-materialize';
import { Link } from 'react-router-dom';
import EditProfile from '../EditProfile';
import Nav from '../../common/nav';
import FollowButton from '../../Follow/FollowButton';
import FollowingList from '../../Follow/FollowingList';
import FollowersList from '../../Follow/FollowersList';

export const profile = props => {
  const { profile } = props;
  const username = profile.username;

  return (
    <>
      <Nav />
      <div className="row">
        <div className="col s1" />
        <div className="col s10">
          <div className="profile">
            <div className="card profile">
              <div className="card-content">
                <div className="row">
                  <div className="col s2">
                    <div />
                    <div className="profile_image">
                      <img
                        src={profile.image}
                        className="responsive-img circle"
                        height="200px"
                        width="200px"
                        alt="Avatar"
                      />
                    </div>
                  </div>
                  <div className="col s8">
                    <div className="username">{profile.username}</div>
                    <Row>
                      <Col s={2}>
                        <span>
                          <span className="number">0</span>
                          Articles
                        </span>
                      </Col>
                      <Col s={2}>
                        <FollowingList {...profile} />
                      </Col>
                      <Col s={2}>
                        <FollowersList {...profile} />
                      </Col>
                    </Row>

                    <div>Bio</div>
                    <div className="bio">{profile.bio}</div>
                  </div>
                  <div className=" col s2">
                    <EditProfile {...profile} />
                    <FollowButton {...profile} />
                  </div>
                </div>
              </div>
              <div className="card-action">
                <a href="/">Articles</a>
                <a href="/">Highlights</a>
                <Link
                  to={{
                    pathname: `/${username}/followers`,
                    username: { username }
                  }}
                >
                  followers
                </Link>
                <Link
                  to={{
                    pathname: `/${username}/following`,
                    username: { username }
                  }}
                >
                  following
                </Link>

                <a href="/">Ratings</a>
              </div>
            </div>
          </div>
        </div>
        <div className="col s1" />
      </div>
    </>
  );
};

export default profile;
