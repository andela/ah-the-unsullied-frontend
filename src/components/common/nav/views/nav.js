import React from 'react';
import { NavLink } from 'react-router-dom';

const NavigationView = () =>{
    return(
      <div>
        <NavLink to='/'>Home</NavLink> 
        <NavLink to='/login'>Login</NavLink>
      </div>
    );
};

export default NavigationView;
