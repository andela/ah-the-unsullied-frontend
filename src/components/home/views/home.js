import React from 'react';

const HomeView = () => {
  return (
    <div>
      <h1>{'Welcome to Author\'s Haven'}</h1>
    </div>
  );
};

export default HomeView;
