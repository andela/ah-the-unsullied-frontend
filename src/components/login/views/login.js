import React from "react";
import { Link } from "react-router-dom";

const LoginView = () => {
  return (
    <form>
      <h1> Login </h1>
      <input type="text" name="email" />
      <br />
      <input type="password" name="password" />
      <br />
      <Link to="/">
        <button type="submit">Login</button>
      </Link>
    </form>
  );
};

export default LoginView;
