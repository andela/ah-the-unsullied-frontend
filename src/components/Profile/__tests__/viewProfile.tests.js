import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16/build';
import { profile } from '../views/viewProfileView';

Enzyme.configure({ adapter: new Adapter() });
describe('get profile', () => {
  let props;
  let wrapper;
  let wrapperInstance;
  let state;
  beforeEach(() => {
    props = {
      auth: {
        isAuthenticated: true,
        user: {
          username: 'nesh'
        }
      },

      bio: 'Bio updated',
      image: 'hssggsgs',

      profile: {
        profile: {}
      },

      match: {
        params: {
          username: 'nesh'
        }
      },

      getUserProfile: async () => {
        await jest.fn();
      }
    };

    state = {
      auth: {},
      profile: {},
      fetched: true
    };

    wrapper = shallow(<profile {...props} />);
    wrapperInstance = wrapper.instance();
  });


  it('should render article view component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
