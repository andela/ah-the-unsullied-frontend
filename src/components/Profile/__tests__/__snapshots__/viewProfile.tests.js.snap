// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get profile should render article view component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <profile
    auth={
      Object {
        "isAuthenticated": true,
        "user": Object {
          "username": "nesh",
        },
      }
    }
    bio="Bio updated"
    getUserProfile={[Function]}
    image="hssggsgs"
    match={
      Object {
        "params": Object {
          "username": "nesh",
        },
      }
    }
    profile={
      Object {
        "profile": Object {},
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "auth": Object {
        "isAuthenticated": true,
        "user": Object {
          "username": "nesh",
        },
      },
      "bio": "Bio updated",
      "getUserProfile": [Function],
      "image": "hssggsgs",
      "match": Object {
        "params": Object {
          "username": "nesh",
        },
      },
      "profile": Object {
        "profile": Object {},
      },
    },
    "ref": null,
    "rendered": null,
    "type": "profile",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "auth": Object {
          "isAuthenticated": true,
          "user": Object {
            "username": "nesh",
          },
        },
        "bio": "Bio updated",
        "getUserProfile": [Function],
        "image": "hssggsgs",
        "match": Object {
          "params": Object {
            "username": "nesh",
          },
        },
        "profile": Object {
          "profile": Object {},
        },
      },
      "ref": null,
      "rendered": null,
      "type": "profile",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
